docs/lib/datastore/fragments/ios/auth-model.md

```diff
@@ -0,0 +1,104 @@
+<amplify-callout>
+When using the `@auth` directive with DataStore, currently only following fields are supported:
+
+1. `allow` with "owner" and "groups"
+2. `ownerField`
+3. `groups`
+4. `operations`
+
+</amplify-callout>
+
+## Updated schema
+
+Create a Model that is read-only.
+
+```graphql
+type Todo
+    @model
+    @auth(rules: [
+        { allow: owner, ownerField: "owner", operations: [create, update, delete] },
+    ]) {
+    id: ID!
+    content: String!
+    owner: String
+}
+```
+
+The API will allow users to query, list, and subscribe to model creation, updates, deletions by other users. A user can only update and delete their own model, that is a model can not be updated or deleted by another user.
+
+## API with Cognito User Pool auth
+
+With the schema, make sure you have selected Cognito User Pool as the auth
```

Comment:
It's unclear why amazon cognito user pools is required here.  I would imagine they could use a federated id, that maps to a cognito user pool... but if the reason to select  amazon cognito user pool here is for simplicity or specific to this example, we should call that out.

Reply:
this one is difficult, although technically they can probably use any OIDC provider, by suggesting so, we don't have a good example or probably even tried this out. and if we try it out, probably going to run into some issue related to the tight coupling. (AppSync has a specific auth mode that is Cognito User Pool) but i get your point and it's worth noting how Cognito's user pool is used in general

Reply:
Per another comment, i think we can just remove calling out that we have to have cognito user pool as auth.